### PR TITLE
Update for Twisted 20.3.0

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This updates the compat testing helpers to work with latest Twisted that support Python 2.7.

We were using a very old Twisted version which was a mix of 12.1 and 15.1

With this change,  any future upgrade to Twisted should not require and update to compat.

Changes
=======

Update the test case.

Move `callCleanup` code fix from chevah/server 

As a drive by, move the static checks to GitHub Actions.
I have also removed the custom skip-ci code as GHA has native support now.
I have also update the brink deps... just in case.

How to try and test the changes
===============================

reviewers: @danuker 

FYI.

This will be used in a PR in chevah/server where  will implement Twisted update.

### CODE REVIEW

**Comment by adiroiban** [comment]
I think that this is ready for a first  needs-review

I still need to fix static-checks now that they are moved to GitHub ...and fix AIX 7.1... as we use the new VM

File: chevah/compat/testing/testcase.py [hunk 1]
```diff
@@ -1006,9 +1009,16 @@ def addCleanup(self, function, *args, **kwargs):
...
+
+        If a cleanup fails, the next cleanups will continue to be called and
+        the first failure is raised.
         """
         for function, args, kwargs in reversed(self.__cleanup__):
-            function(*args, **kwargs)
+            try:
+                function(*args, **kwargs)
+            except Exception as error:  # noqa:cover
+                self._teardown_errors.append(error)
```
Comment (danuker) [inline chevah/compat/testing/testcase.py:1019]: How about appending a `traceback.format_exc()` instead of the error itself?

**Review by danuker — APPROVED** [review]
I approve in general; but I would like the tracebacks of the errors also. Do you think they would be too noisy?

File: chevah/compat/testing/testcase.py (hunk 1 above)
Comment (adiroiban) [reply chevah/compat/testing/testcase.py:1019]: True. Thanks.

**Comment by dumol** [comment]
As I'm trying to get rid of Alpine Linux VMs, have added a workflow to test it under Docker.

As I have found more Buildbot OS'es were being required from Buildbot (but missing because they were recenlty removed), I've added them back through Docker: Amazon Linux 2, CentOS 7/8.

While at it, I've also added Amazon Linux 2018.03 and CentOS 8.2.

Some tests fail consistently on CentOS 8.x (and they are not the same on 8,2 and latest 8).

**Comment by adiroiban** [comment]
Thanks @dumol for the update. This is a great start.

Maybe some tests are failing due to umask differences.

But I think that the remaining can be fixed as part of the py3 migration.

**Pull request merged**